src/lib/bcmp/bcmp_ping.cpp

```diff
@@ -20,23 +20,23 @@ static uint16_t _expected_payload_len = 0;
   \param *addr - ip address to send to send ping request to
   \param *payload - payload to send with ping
   \param payload_len - length of payload to send
-  \ret ERR_OK if successful
+  \ret BmOK if successful
 */
-err_t bcmp_send_ping_request(uint64_t node_id, const ip_addr_t *addr, const uint8_t *payload,
+BmErr bcmp_send_ping_request(uint64_t node_id, const void *addr, const uint8_t *payload,
                              uint16_t payload_len) {
 
   if (payload == NULL) {
     payload_len = 0;
   }
 
-  uint16_t echo_len = sizeof(bcmp_echo_request_t) + payload_len;
+  uint16_t echo_len = sizeof(BcmpEchoRequest) + payload_len;
 
-  uint8_t *echo_req_buff = static_cast<uint8_t *>(pvPortMalloc(echo_len));
-  configASSERT(echo_req_buff);
+  uint8_t *echo_req_buff = static_cast<uint8_t *>(bm_malloc(echo_len));
```

Comment:
I think we should try to C-ify these files as much as possible seeing as all of bm_core will be C based. Could you change the `static_cast`/other C++ casts to C casts? (`uint8_t *`).